src/modules/uORB/uORBDevices_nuttx.cpp

```diff
@@ -669,18 +673,12 @@ uORB::DeviceMaster::ioctl(struct file *filp, int cmd, unsigned long arg)
 					*(adv->instance) = group_tries;
 				}
 
-				/* driver wants a permanent copy of the node name, so make one here */
-				objname = strdup(meta->o_name);
-
-				if (objname == nullptr) {
-					return -ENOMEM;
-				}
+				objname = meta->o_name; //no need for a copy, meta->o_name will never be freed or changed
```

Comment:
Yes, both objname & o_name are const, however look at the definition of `strdup`:
```
The strdup() function shall return a pointer to a new string, which is a duplicate of the string pointed to by s1. The returned pointer can be passed to free()
```
http://pubs.opengroup.org/onlinepubs/009695399/functions/strdup.html

So even if the compiler knew the sematics of `strdup`, it could not avoid the dynamic allocation, because the new string can be passed to free.